File: generator/src/java/darkness/generator/scripts/uka15/DemoSubScript.java
package darkness.generator.scripts.uka15;

import darkness.generator.api.BulbGroup;
import darkness.generator.api.BulbRGB;

public class DemoSubScript extends BaseScript {
	@Override
	public void run() {
		super.run();
		BulbRGB previous = null;
		for (BulbGroup letter : letters) {
			for (BulbRGB bulb : letter) {
				if (previous != null) {
					relinquish(previous);
				}
				set(bulb, 255, 0, 0);
				previous = bulb;
				next();
			}
		}
		relinquish(previous);
	}
}

File: generator/src/kotlin/darkness/generator/scripts/uka15/DemoSubScript.kt
package darkness.generator.scripts.uka15

import darkness.generator.api.BulbRGB

class DemoSubScript : BaseScript() {
    override suspend fun run() {
        super.run()
        for (letter in letters) {
            for (bulb in letter) {
                set(bulb, 255, 0, 0)
                next()
            }
        }
    }
}
